plugins/cassandra/cassandrawu.cpp

```diff
@@ -3528,9 +3528,37 @@ class CCasssandraWorkUnitFactory : public CWorkUnitFactory, implements ICassandr
             case WUStateDebugRunning:
             case WUStateBlocked:
             case WUStateAborting:
-                SessionId agent = getUnsignedResult(NULL, cass_row_get_column(row, 1));
-                if (agent && checkAbnormalTermination(wuid, state, agent))
-                    return state;
+                if (queryDaliServerVersion().compare("2.1")>=0)
+                {
+                    SessionId agent = getUnsignedResult(NULL, cass_row_get_column(row, 1));
+                    if(agent && querySessionManager().sessionStopped(agent, 0))
+                    {
+                        // MCK - does checking state after sessionStopped reveal meaningful info ?
```

Comment:
I think it does, but I think this code would be simpler if it noted stopped and looped (so used existing state loading/checking code), then actually reported abnormal termination if state still in 'bad' state, i.e. on next pass.

Same as : https://github.com/hpcc-systems/HPCC-Platform/pull/8963/files#r74224306